zetta_utils/cloud/resource_allocation/k8s/job.py

```diff
@@ -0,0 +1,143 @@
+    selector: Optional[k8s_client.V1LabelSelector] = None,
+    suspend: Optional[bool] = False,
+):
+    pod_template = k8s_client.V1PodTemplateSpec(metadata=meta, spec=pod_spec)
+    return k8s_client.V1JobSpec(
+        active_deadline_seconds=active_deadline_seconds,
+        backoff_limit=backoff_limit,
+        selector=selector,
+        suspend=suspend,
+        template=pod_template,
+    )
+
+
+def get_job_template(
+    name: str,
+    pod_spec: k8s_client.V1PodSpec,
+    active_deadline_seconds: Optional[int] = None,
+    backoff_limit: Optional[int] = 3,
+    labels: Optional[Dict[str, str]] = None,
+    selector: Optional[k8s_client.V1LabelSelector] = None,
+    suspend: Optional[bool] = False,
+) -> k8s_client.V1JobTemplateSpec:
+    meta = k8s_client.V1ObjectMeta(name=name, labels=labels)
+    job_spec = _get_job_spec(
+        pod_spec=pod_spec,
+        meta=meta,
+        active_deadline_seconds=active_deadline_seconds,
+        backoff_limit=backoff_limit,
+        selector=selector,
+        suspend=suspend,
+    )
+    return k8s_client.V1JobTemplateSpec(metadata=meta, spec=job_spec)
+
+
+def get_job(
+    name: str,
+    pod_spec: k8s_client.V1PodSpec,
+    active_deadline_seconds: Optional[int] = None,
+    backoff_limit: Optional[int] = 3,
+    labels: Optional[Dict[str, str]] = None,
+    selector: Optional[k8s_client.V1LabelSelector] = None,
+    suspend: Optional[bool] = False,
+) -> k8s_client.V1Job:
+    meta = k8s_client.V1ObjectMeta(name=name, labels=labels)
+    job_spec = _get_job_spec(
+        pod_spec=pod_spec,
+        meta=meta,
+        active_deadline_seconds=active_deadline_seconds,
+        backoff_limit=backoff_limit,
+        selector=selector,
+        suspend=suspend,
+    )
+    return k8s_client.V1Job(metadata=meta, spec=job_spec)
+
+
+@contextmanager
+def job_ctx_manager(
+    execution_id: str,
+    cluster_info: ClusterInfo,
+    job: k8s_client.V1Job,
+    secrets: List[k8s_client.V1Secret],
+    namespace: Optional[str] = "default",
+):
+    configuration, _ = get_cluster_data(cluster_info)
+    k8s_client.Configuration.set_default(configuration)
+
+    batch_v1_api = k8s_client.BatchV1Api()
+    with secrets_ctx_mngr(execution_id, secrets, cluster_info):
+        logger.info(f"Creating k8s job `{job.metadata.name}`")
+        batch_v1_api.create_namespaced_job(body=job, namespace=namespace)
+        register_execution_resource(
+            ExecutionResource(
+                execution_id,
+                ExecutionResourceTypes.K8S_JOB.value,
+                job.metadata.name,
+            )
+        )
+
+        try:
+            yield
+            while True:
+                job = batch_v1_api.read_namespaced_job_status(
+                    name=job.metadata.name,
+                    namespace=namespace,
+                )
+                logger.info(f"`{job.metadata.name}` job status: {job.status.ready}")
+                if job.status.ready == 1:
```

Comment:
Is ready a boolean? If it is, this should be `if job.status.ready:`

Reply:
no it's `int`